packages/sdk-codegen/src/swift.gen.spec.ts

```diff
@@ -87,6 +87,42 @@ public enum PermissionType: String, Codable {
      * A spaced out property name (read-only)
      */
     public var computation_time: Float?
+
+    public init(project_name: String? = nil, project_digest: String? = nil, computation_time: Float? = nil) {
+        self.project_name = project_name
+        self.project_digest = project_digest
+        self.computation_time = computation_time
+    }
+
+}`
+      expect(actual).toEqual(expected)
+    })
+  })
+
+  describe('constructor', () => {
+    it('generates public inits with required args', () => {
+      const type = apiTestModel.types.EmbedParams
+      const actual = gen.declareType(indent, type)
+      const expected = `public struct EmbedParams: SDKModel {
+    /**
+     * The complete URL of the Looker UI page to display in the embed context. For example, to display the dashboard with id 34, \`target_url\` would look like: \`https://mycompany.looker.com:9999/dashboards/34\`. \`target_uri\` MUST contain a scheme (HTTPS), domain name, and URL path. Port must be included if it is required to reach the Looker server from browser clients. If the Looker instance is behind a load balancer or other proxy, \`target_uri\` must be the public-facing domain name and port required to reach the Looker instance, not the actual internal network machine name of the Looker instance.
+     */
+    public var target_url: URI
+    /**
+     * Number of seconds the SSO embed session will be valid after the embed session is started. Defaults to 300 seconds. Maximum session length accepted is 2592000 seconds (30 days).
+     */
+    public var session_length: Int64?
+    /**
+     * When true, the embed session will purge any residual Looker login state (such as in browser cookies) before creating a new login state with the given embed user info. Defaults to true.
+     */
+    public var force_logout_login: Bool?
+
+    public init(_ target_url: URI, session_length: Int64? = nil, force_logout_login: Bool? = nil) {
+        self.target_url = target_url
+        self.session_length = session_length
+        self.force_logout_login = force_logout_login
+    }
```

Comment:
Note that required parameters are positional rather than named. If this is not desirable, we can update so all parameters must be named.